From 69b582d7da24fabb191061c26764ab684544d3d4 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 19 Jan 2022 07:27:16 +0200
Subject: [PATCH] [SW-58071] arch: common: remove unused functions

Since regmap and regmask headers are used in common function to
get register names, this patch removes functions which were used to
return register offset due to different register names in greco and
gaudi2.

Change-Id: I904f86a3cd1cb6c9065ad1ffeb4fcd55a804d080
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 6a6e784..3a2486c 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -48,19 +48,9 @@
 	regmap_writel(reg, (mmPSOC_SECURITY_SW_STATUS_0 + offset  * 4));
 }
 
-uint32_t get_efuse_read_reg(void)
-{
-	return mmPSOC_EFUSE_READ_CTRL;
-}
-
 uint32_t get_mem_cfg_mask(void)
 {
 	return PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_MEM_SIZE_MASK;
-}
-
-uint32_t get_fld_size_shift(void)
-{
-	return PSOC_GLOBAL_CONF_SPI_DMA_CTRL_MEM_SIZE_SHIFT;
 }
 
 void update_warm_reset_config(void)
diff --git a/arch/greco/arc_elf32/common.c b/arch/greco/arc_elf32/common.c
index b409f6b..9c034d9 100644
--- a/arch/greco/arc_elf32/common.c
+++ b/arch/greco/arc_elf32/common.c
@@ -18,19 +18,9 @@
 #define IATU_LWR_TARGET_ADDR_OFF_INBOUND	(IATU_BASE + 0x114)
 #define IATU_UPPER_TARGET_ADDR_OFF_INBOUND	(IATU_BASE + 0x118)
 
-uint32_t get_efuse_read_reg(void)
-{
-	return mmPSOC_EFUSE_READ;
-}
-
 uint32_t get_mem_cfg_mask(void)
 {
 	return PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_FLD_SIZE_MASK;
-}
-
-uint32_t get_fld_size_shift(void)
-{
-	return PSOC_GLOBAL_CONF_SPI_DMA_CTRL_FLD_SIZE_SHIFT;
 }
 
 void update_warm_reset_config(void)
diff --git a/common/common.h b/common/common.h
index ce940ce..3e2d944 100644
--- a/common/common.h
+++ b/common/common.h
@@ -9,7 +9,6 @@
 uint32_t get_mem_cfg_mask(void);
 uint8_t get_sris_mode_from_strap(void);
 void enable_flr_handling_fsm(void);
-uint32_t get_fld_size_shift(void);
 void update_warm_reset_config(void);
 void arcboot_rom_handle_tpm(void);
 void arcboot_rom_request_cpld_wdog(void);
diff --git a/common/efuse.h b/common/efuse.h
index 08c71ed..e3f6aea 100644
--- a/common/efuse.h
+++ b/common/efuse.h
@@ -36,6 +36,5 @@
 int fuse_read(uint32_t word, uint32_t *val);
 uint8_t efuse_otp_get_fw_version_security(void);
 void efuse_get_hash(uint32_t *key_hash);
-uint32_t get_efuse_read_reg(void);
 
 #endif /* _EFUSE_H_ */
